test/testutils/syncenginetestutils.cpp

```diff
@@ -101,13 +115,106 @@ void DiskFileModifier::mkdir(const QString &relativePath)
 
 void DiskFileModifier::rename(const QString &from, const QString &to)
 {
-    QVERIFY(_rootDir.exists(from));
-    QVERIFY(_rootDir.rename(from, to));
+//    QVERIFY(_rootDir.exists(from));
+//    QVERIFY(_rootDir.rename(from, to));
+
+    _processArguments.append({"rename", from, to });
 }
 
 void DiskFileModifier::setModTime(const QString &relativePath, const QDateTime &modTime)
 {
-    OCC::FileSystem::setModTime(_rootDir.filePath(relativePath), OCC::Utility::qDateTimeToTime_t(modTime));
+//    OCC::FileSystem::setModTime(_rootDir.filePath(relativePath), OCC::Utility::qDateTimeToTime_t(modTime));
+
+    _processArguments.append({"mtime", relativePath, QString::number(modTime.toSecsSinceEpoch()) });
+}
+
+void DiskFileModifier::incModTime(const QString &relativePath, int secondsToAdd)
+{
+    time_t mtime = OCC::FileSystem::getModTime(_rootDir.filePath(relativePath));
+    auto newMTime = OCC::Utility::qDateTimeFromTime_t(mtime).addSecs(secondsToAdd);
+    setModTime(relativePath, newMTime);
+}
+
+class HelperProcess : public QProcess
+{
+public:
+    HelperProcess(const QDir &rootDir, QStringList processArguments) // copy is on purpose
+    {
+        if (processArguments.isEmpty()) { // fast-path:
+            finished = true;
+            succeeded = true;
+            return;
+        }
+
+        setProcessChannelMode(QProcess::MergedChannels);
+        processArguments.prepend(rootDir.absolutePath());
+        QObject::connect(this, &QProcess::readyRead, [this]() {
+            while (canReadLine()) {
+                qDebug() << "helper output:" << readLine();
+            }
+        });
+        QObject::connect(this, QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished),
+            [this](int exitCode, QProcess::ExitStatus exitStatus) {
+                qDebug() << "helper finished:" << exitCode << exitStatus;
+                this->finished = true;
+                this->succeeded = exitStatus == QProcess::NormalExit && exitCode == 0;
+            });
+
+        qDebug() << "Starting helper:" << TEST_HELPER_EXE << processArguments;
+        start(TEST_HELPER_EXE, processArguments);
+    }
+
+public:
+    bool finished = false;
+    bool succeeded = true;
+};
+
+bool DiskFileModifier::applyModifications(FakeFolder &ff, OCC::Vfs::Mode mode)
+{
+    if (mode == OCC::Vfs::Off) { // Classic mode:
+        HelperProcess helper(_rootDir, _processArguments);
+        _processArguments.clear(); // the helper has a copy, we clean it now for the next run
+        while (!helper.finished) {
+            QThread::currentThread()->eventDispatcher()->processEvents(QEventLoop::AllEvents);
+            helper.waitForFinished();
+        }
+        if (!helper.succeeded) {
+            return false;
+        }
+        return ff.syncOnce();
```

Comment:
Hm the implicit syncOnce will result in local==remote state, if nothing goes wrong. Is that really what we want here?